control-plane-operator/controllers/hostedcontrolplane/etcd/reconcile.go

```diff
@@ -183,7 +183,34 @@ func buildEnsureDNSContainer(p *EtcdParams, ns string) func(c *corev1.Container)
 
 func buildEtcdContainer(p *EtcdParams, namespace string) func(c *corev1.Container) {
 	return func(c *corev1.Container) {
-		script := `
+		var script string
+
+		if p.IPv6 {
+			script = `
+/usr/bin/etcd \
+--data-dir=/var/lib/data \
+--name=${HOSTNAME} \
+--initial-advertise-peer-urls=https://${HOSTNAME}.etcd-discovery.${NAMESPACE}.svc:2380 \
+--listen-peer-urls=https://[${POD_IP}]:2380 \
+--listen-client-urls=https://[${POD_IP}]:2379,https://localhost:2379 \
+--advertise-client-urls=https://${HOSTNAME}.etcd-client.${NAMESPACE}.svc:2379 \
+--listen-metrics-urls=https://[::]:2382 \
+--initial-cluster-token=etcd-cluster \
+--initial-cluster=${INITIAL_CLUSTER} \
+--initial-cluster-state=new \
+--quota-backend-bytes=${QUOTA_BACKEND_BYTES} \
+--snapshot-count=10000 \
+--peer-client-cert-auth=true \
+--peer-cert-file=/etc/etcd/tls/peer/peer.crt \
+--peer-key-file=/etc/etcd/tls/peer/peer.key \
+--peer-trusted-ca-file=/etc/etcd/tls/etcd-ca/ca.crt \
+--client-cert-auth=true \
+--cert-file=/etc/etcd/tls/server/server.crt \
+--key-file=/etc/etcd/tls/server/server.key \
+--trusted-ca-file=/etc/etcd/tls/etcd-ca/ca.crt
+`
```

Comment:
the only different thing here is the `--listen-metrics-urls` right?
can we just substitute this using a variable instead of copying the duplicating the whole script using? something like 
```go
metricsUrl := "https://0.0.0.0:2382"
if ipv6 {
  metricsUrl = "https://[::]:2382"
}

scriptTemplate := `
...
--listen-metrics-urls=%s \
...
`

script := fmt.Sprintf(scriptTemplate, metricsUrl)
```

Reply:
I've updated the whole script and metrics script to work as a template.

Also added the unitTests for Metrics (I've forgot that)